### PR TITLE
Fix auto-config implementation

### PR DESCRIPTION
**Pull request by banasraf**
This PR adds a top-level auto-config call `AutofillConfig` and fixes the implementation.
All of the autofill calls were refactored to modify the provided JSON values in-place instead of producing new values.

It also fixes a typo in a field name "allow_ragged_batches" -> "allow_ragged_batch"

### CODE REVIEW

**Review by szalpal — DISMISSED** [review]

File: src/utils/triton.h [hunk 1]
```diff
@@ -155,6 +155,13 @@ class TritonError : public UniqueHandle<TRITONSERVER_Error *, TritonError>, publ
       return "";
     }
   }
+
+  TritonError &operator=(TRITONSERVER_Error *error) {
+    if (handle_ != error) {
+      *this = TritonError(error);
```
[the remark this answers is not in the document]
Comment (banasraf) [reply src/utils/triton.h:161]: fyi: this calls move assign operator inherited from UniqueHandle.

[the remark this answers is not in the document]
Comment (prak-nv) [reply src/utils/triton.h:161]: All should be fine here

File: src/config_tools/config_tools.cc [hunk 2]
```diff
@@ -364,12 +380,16 @@ void ValidateOutputs(TritonJson::Value &outs, const std::vector<IOConfig> &out_c
 
 int ReadMaxBatchSize(TritonJson::Value &config) {
   int64_t bs = -1;
-  config.MemberAsInt("max_batch_size", &bs);
+  TritonError{config.MemberAsInt("max_batch_size", &bs)};
   if (bs > std::numeric_limits<int>::max() || bs < -1) {
```
Comment (prak-nv) [inline src/config_tools/config_tools.cc:384]: Just confirming: -1 and 0 shouldn't really throw?

**Review by prak-nv — APPROVED** [review]

File: src/config_tools/config_tools.cc (hunk 2 above)
Comment (banasraf) [reply src/config_tools/config_tools.cc:384]: Yeah, -1 and 0 are equivalent to missing field - in such situation we want to return -1. Less than -1 must mean  a user mistake.

**Pull request merged**